Source/VolunteerReporting/TextMessaging/TextMessageProcessor.cs

```diff
@@ -48,131 +47,83 @@ public class TextMessageProcessor : ICanProcessTextMessage
         public void Process(TextMessage message)
         {
             var parsingResult = _textMessageParser.Parse(message);
+            var isTextMessageFormatValid = parsingResult.IsValid;
             var dataCollector = _dataCollectors.GetByPhoneNumber(message.OriginNumber);
+            var unknownDataCollector = dataCollector == null;
             var caseReportId = Guid.NewGuid();
             var caseReporting = _caseReportingRepository.Get(caseReportId);
```

Comment:
This class looks fine, but two minor things that botter me:
1) I would prefer to have private methods in the class to handle the different cases, instead of having to use the return statements. (Minor thing really)
2) Shouldn't we here instead of calling on the AggregateRoot for CaseReport emit a command to a CaseReport command handler? 

Other than that, it look fine. An improvement for sure.

Reply:
1. Interesting thought. I tried to do something similar, but ended up with this since it is easy enough to follow. How would you write the private methods that would ensure that only one case report was emited?
2. We have gone over this before and we are still not quite sure how this should be. The reason that we do it this way, is that sending an SMS can be seen as the command. If we recieve the SMS, that is a fact, something that happend. As you can see, this logic happens in a Process method, which means that we cannot recject it based on i.e. input validation. If the SMS is wrong, we must handle it by emiting the corresponding event. Make sense?

Reply:
1) I have no concrete suggestion at the moment, I just stop and think when I see multiple return statements in a method.

2) I see your point, Bjørn. It may be beneficial to look at textmessage processing as it's own thing entirely even.